cdap-app-fabric/src/main/java/co/cask/cdap/app/runtime/ProgramController.java

```diff
@@ -109,9 +109,7 @@
 
     State(ProgramRunStatus runStatus) {
       this.runStatus = runStatus;
-      this.programStatus = ProgramRunStatus.STARTING == runStatus || ProgramRunStatus.RUNNING == runStatus ?
```

Comment:
Yes, it shouldn't be removed. I was thinking that we could now make the definition of when a program runs a bit more granular but we can't do that as it would require changes to the `ProgramStatus` enum here.